### PR TITLE
:sparkles: Add `flatcar-sysext` template to use regular Flatcar images

### PR DESCRIPTION
**Pull request by tormath1**
**What this PR does / why we need it**: In this PR, we add a new template: `flatcar-sysext`. This template has two main benefits:
* consume upstream images (directly from Flatcar release servers without going through the `image-builder` requirement)
* automatically download and install Kubernetes patch releases (disabled by default for now to mimic the current CAPI/CAPO behavior)

**Special notes for your reviewer**: This can solve the question of providing CAPO image for Flatcar - as a user can directly consume the upstream image.

Some resources:
* CAPO demo during the office hours: https://www.youtube.com/watch?v=RwKl-PtSjjw
* For more context, here's a demo I did a few months ago during the Flatcar office hours: https://www.youtube.com/watch?v=omppbFNxSDU&t=210s
* https://github.com/flatcar/sysext-bakery/ (where the sysext images are produced and stored)
* https://www.flatcar.org/docs/latest/provisioning/sysext/
* https://www.freedesktop.org/software/systemd/man/latest/systemd-sysext.html
* https://www.freedesktop.org/software/systemd/man/latest/systemd-sysupdate.html

**TODOs**:

- [x] squashed commits
- if necessary:
  - [x] includes documentation
  - [x] adds unit tests

/hold

### CODE REVIEW

**Review by mdbooth — APPROVED** [review]
I've left some comments, but as you're our flatcar maintainer I'm happy for you to merge this whenever you see fit.

File: hack/ci/cloud-init/controller.yaml.tpl
Comment (EmilienM) [inline hack/ci/cloud-init/controller.yaml.tpl:68]: I don't have enough experience to tell if that's ok to include an image from another endpoint than other images in the list. I'm sure there is a good reason to have them stored on google vs others. I'll let the maintainers to comment on this one.

**Pull request merged**
